src/api/posts.ts

```diff
@@ -1,11 +1,25 @@
-import { getRequest } from './api';
+import { Post } from '../react-app-env';
 
-export const getAllPosts = async (): Promise<Post[]> => getRequest('/posts');
+export const BASE_URL = 'https://mate.academy/students-api/';
 
-export const getUserPosts = async (userId: number): Promise<Post[]> => (
-  getRequest(`/posts?userId=${userId}`)
-);
+export async function getPosts(userId: string): Promise<Post[]> {
+  const response = await fetch(`${BASE_URL}/posts?userId=${userId}`);
 
-export const getPostDetails = async (postId: number): Promise<Post> => (
-  getRequest(`/posts/${postId}`)
-);
\ No newline at end of file
+  if (!response.ok) {
+    throw new Error(`${response.status} - ${response.statusText}`);
+  }
+
+  return response.json();
+}
+
+export async function getPostbyId(postId:number) {
+  const response = await fetch(`${BASE_URL}/posts/${postId}`);
+
+  return response.json();
+}
+
+export async function getAllPosts() {
+  const response = await fetch(`${BASE_URL}/posts`);
```

Comment:
you can simplify this string as follows 
return await fetch(`${BASE_URL}/posts`);